File: frontend/src/types/prediction.ts
export interface CandidateNumbers {
  red_numbers: number[];
  blue_number: number;
  score?: number;
  red_hit_count?: number | null;
  blue_hit?: boolean | null;
  prize_level?: string | null;
}

export interface PredictionRunRequest {
  target_issue_no: string;
  train_until_issue_no: string;
  model_keys?: string[];
  candidate_strategy?: string;
  candidate_count?: number;
}

export interface ModelPrediction {
  model_key: string;
  target_issue_no: string;
  train_until_issue_no: string;
  draw_datetime?: string | null;
  red_probabilities: Record<number, number>;
  blue_probabilities: Record<number, number>;
  candidate_numbers: CandidateNumbers[];
  prediction_run_id?: number | null;
  run_type?: string | null;
  best_red_hit_count?: number | null;
  best_blue_hit?: boolean | null;
  evaluated_at?: string | null;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class CandidateNumbers(BaseModel):
    red_numbers: list[int]
    blue_number: int
    score: float | None = None
    red_hit_count: int | None = None
    blue_hit: bool | None = None
    prize_level: str | None = None


class PredictionRunRequest(BaseModel):
    target_issue_no: str
    train_until_issue_no: str
    model_keys: list[str] | None = None
    candidate_strategy: str = "top_k"
    candidate_count: int = 5


class ModelPredictionRead(BaseModel):
    model_key: str
    target_issue_no: str
    train_until_issue_no: str
    draw_datetime: str | None = None
    red_probabilities: dict[int, float]
    blue_probabilities: dict[int, float]
    candidate_numbers: list[CandidateNumbers]
    prediction_run_id: int | None = None
    run_type: str | None = None
    best_red_hit_count: int | None = None
    best_blue_hit: bool | None = None
    evaluated_at: str | None = None
